xdsl/dialects/csl.py

```diff
@@ -80,6 +82,82 @@ class ModuleKind(StrEnum):
     PROGRAM = "program"
 
 
+class TaskKind(StrEnum):
+    LOCAL = "local"
+    DATA = "data"
+    CONTROL = "control"
+
+
+class FuncBase:
```

Comment:
We can make this module private, right? It's not supposed to be used by anyone except `TaskOp` and `FuncOp`.